### ISSUE TITLE
Move pineko compare here

### ISSUE THREAD

**Issue by felixhekhorn** [N3PDF/pineappl#112]
Since we were thinking to dissolve `pineko` and to merge it with `fkutils` I was wondering whether we could move the compare script https://github.com/N3PDF/pineko/blob/main/src/pineko/comparator.py inside `PineAPPL` - more specific into `pineappl diff` since the functionality is very similar. The only addition is the restriction of orders onto the process level grid, something that proved very useful in the context of https://github.com/NNPDF/fktables/issues/16

and I would like `diff` to retain the binning information (x,Q2 limits etc.) (which I believe at the moment he is not doing) (maybe by a flag?)

**Comment by cschwan** [comment]
The binning information is printed, but I can add a flag that selects a specific subset of orders. For your purposes, the comparison of FK tables and PineAPPL grids, the flag `--ignore-orders` should do the trick. Can you confirm that?

**Comment by cschwan** [comment]
@felixhekhorn This is clearly a bug, because if we take the first output seriously, both grids are exactly the same - which they clearly aren't.

**Comment by cschwan** [comment]
@felixhekhorn It's not because it's the EW coupling `alpha` and not the QCD `alphas`!

**Comment by cschwan** [comment]
@felixhekhorn In commit df89a23344a4ba523fe2de40ff896de2a8d5ab76 I changed the behaviour of `pineappl diff` slightly to fit our needs. Could you have a look at it? It should cover at least two cases:

1) compare a converted APPLgrid/fastNLO table with a PineAPPL grid generated from us: `pineappl diff --ignore-lumis --orders2=a2,a2as1 appl-converted.pineappl mg5-generated.pineappl`. In that case the luminosities and orders will be different because EW corrections are present in our grids. We therefore must ignore the difference in the lumis (`--ignore-lumis`), and from our second grid we select the highest orders in QCD with `--orders2=a2,a2as1`. In practice this doesn't always work yet because the converted grid has the wrong alpha order.
2) compare an FK table with another grid, possibly excluding EW corrections: `pineappl diff --ignore-lumis --ignore-orders --orders2=a2,a2as1 fk-table.pineappl mg5-generated.pineappl`. Here we need to add `--ignore-orders` because they are completely different among the two grids.

**Comment by felixhekhorn** [comment]
Seems alright :+1: close?

**Comment by cschwan** [comment]
> 1. In practice this doesn't always work yet because the converted grid has the wrong alpha order.

This'll be fixed in https://github.com/N3PDF/pineappl/issues/113.

**Comment by felixhekhorn** [comment]
@cschwan I was about to delete the function in `pineko`, but then I realized that it is the only way to envoke that in python - I guess we just keep it where it is, right? the other option would be to move it into `pineappl_py`, but that would introduce `pandas` as a dependency there and that we maybe want to avoid ... cc @AleCandido

**Comment by AleCandido** [comment]
I agree to keep out `pandas`  from `pineapp_py`, let's keep as few dependencies as possible
